diff --git a/PKGBUILD b/PKGBUILD
index 230f88b..6756fad 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -1,7 +1,11 @@
 # Maintainer: Andreas Radke <andyrtr@archlinux.org>
 # Contributor: Sergej Pupykin <pupykin.s+arch@gmail.com>
 
-pkgname=openjpeg2
+pkgbase=openjpeg2
+pkgname=(
+  openjpeg2
+  openjpeg2-docs
+)
 pkgver=2.5.4
 pkgrel=1
 pkgdesc="An open source JPEG 2000 codec, version ${pkgver}"
@@ -10,7 +14,7 @@ license=('BSD-2-Clause' 'MIT')
 url="https://github.com/uclouvain/openjpeg"
 makedepends=('cmake' 'doxygen' 'graphviz')
 depends=('libpng' 'libtiff' 'lcms2' 'glibc')
-source=("$pkgname-$pkgver.tar.gz::https://github.com/uclouvain/openjpeg/archive/v$pkgver.tar.gz")
+source=("$pkgbase-$pkgver.tar.gz::https://github.com/uclouvain/openjpeg/archive/v$pkgver.tar.gz")
 sha256sums=('a695fbe19c0165f295a8531b1e4e855cd94d0875d2f88ec4b61080677e27188a')
 
 prepare() {
@@ -33,7 +37,34 @@ check() {
   ctest --test-dir build --output-on-failure
 }
 
-package() {
+_pick() {
+  local p="$1" f d; shift
+  for f; do
+    d="$srcdir/$p/${f#$pkgdir/}"
+    mkdir -p "$(dirname "$d")"
+    mv "$f" "$d"
+    rmdir -p --ignore-fail-on-non-empty "$(dirname "$f")"
+  done
+}
+
+package_openjpeg2() {
+  optdepends=('openjpeg2-docs: API documentation')
   DESTDIR="$pkgdir" cmake --install build
-  install -Dm 644 "${srcdir}"/openjpeg-"${pkgver}"/LICENSE -t "${pkgdir}"/usr/share/licenses/${pkgname}/  
+
+  (
+    cd "$pkgdir"
+    _pick docs usr/share/doc/openjpeg/html
+  )
+
+  install -Dm644 openjpeg-"${pkgver}"/LICENSE \
+    -t "${pkgdir}"/usr/share/licenses/${pkgname}/
+}
+
+package_openjpeg2-docs() {
+  pkgdesc+=" - documentation"
+
+  mv docs/* "$pkgdir"
+
+  install -Dm644 openjpeg-"${pkgver}"/LICENSE \
+    -t "${pkgdir}"/usr/share/licenses/${pkgname}/
 }
